SDL/macbundle.sh

```diff
@@ -9,24 +9,36 @@ if [ ! -f "${PPSSPPSDL}" ]; then
 fi
 
 SDL=$(otool -L "${PPSSPPSDL}" | grep -v @executable_path | grep -Eo /.+libSDL.+dylib)
-if [ "${SDL}" = "" ]; then
-  echo "SDL is already bundled/unused."
-  exit 0
-fi
+if [ "${SDL}" != "" ]; then
+  if [ ! -f "${SDL}" ]; then
+    echo "Cannot locate SDL: ${SDL}!"
+    exit 0
+  fi
 
-if [ ! -f "${SDL}" ]; then
-  echo "Cannot locate SDL: ${SDL}!"
-  exit 0
-fi
+  echo "Installing SDL from ${SDL}..."
 
-echo "Installing SDL from ${SDL}..."
+  SDLNAME=$(basename "${SDL}")
+  mkdir -p "${PPSSPP}/Contents/Frameworks" || exit 0
+  cp -r "$SDL" "${PPSSPP}/Contents/Frameworks" || exit 0
+  install_name_tool -change "${SDL}" "@executable_path/../Frameworks/${SDLNAME}" "${PPSSPPSDL}" || exit 0
+elif [ "$(otool -L "${PPSSPPSDL}" | grep @rpath/SDL)" != "" ]; then
+  cd "$(dirname "$0")"
+  RPATH="$(pwd)/macOS"
+  cd -
+  SDL="${RPATH}/SDL2.framework"
+  if [ ! -d "${SDL}" ]; then
+    echo "Cannot locate SDL.framework: ${SDL}!"
+    exit 0
+  fi
 
-SDLNAME=$(basename "${SDL}")
-mkdir -p "${PPSSPP}/Contents/Frameworks" || exit 0
-cp -r "$SDL" "${PPSSPP}/Contents/Frameworks" || exit 0
-install_name_tool -change "${SDL}" "@executable_path/../Frameworks/${SDLNAME}" "${PPSSPPSDL}" || exit 0
+  rm -rf "${PPSSPP}/Contents/Frameworks/SDL2.framework" || exit 0
+  mkdir -p "${PPSSPP}/Contents/Frameworks" || exit 0
+  cp -a "$SDL" "${PPSSPP}/Contents/Frameworks" || exit 0
+  install_name_tool -rpath "${RPATH}" "@executable_path/../Frameworks" "${PPSSPPSDL}" || echo "Already patched."
+fi
 
 GIT_VERSION_LINE=$(grep "PPSSPP_GIT_VERSION = " "$(dirname "${0}")/../git-version.cpp")
+echo "Setting version to ${GIT_VERSION_LINE}..."
```

Comment:
I am worried about the version string in the builds. Currently there seem to be 2 bugs:

- The generated version string only has the commit hash and not the version number (1.2.3). See [this print](https://github.com/hrydgard/ppsspp/pull/14839/checks?check_run_id=3581556484#step:8:1323) from the log.
- How can I make sure there is no commit suffix in the tagged releases? I.e. I have the 1.2.3 version name alone if it is a tagged build.

Reply:
I believe fixed (1) in https://github.com/vit9696/ppsspp/commit/ed046aebf5608869fd49fb55204715b01bc5620b. I am not sure what route shall I take to fix (2). Is it alright to make `git-version.cpp` with `PPSSPP_GIT_VERSION_NO_UPDATE` set to `1` in one of the CI stages for tagged commits?

Reply:
I guess that's fine, as long as it has the right values.  I would think you'd need to fetch tags, i.e. `git fetch --tags` or more ideally `git fetch <remote> 'refs/tags/*:refs/tags/*` to be able to have `git describe` fetch the right thing.  That would probably fetch less and be faster.

-[Unknown]

Reply:
I tried that, but it did not work for me. I believe the reason is that we not only need to fetch the tags, but also connect tags to the history, otherwise one would get `ge1cb1c9a6` instead of `v1.11.3-1350-ge1cb1c9a6` for intermediate releases. I changed the code to only fetch history on macOS, however, which should improve the situation quite a bit.